superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx

```diff
@@ -235,6 +249,36 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     }
   };
 
+  const expandForm = (trueOrFalse: boolean) => {
+    // Count starts a 0 (closed)
+    // Any checkbox in this for that gets checked
+    // will increase the count and keep the form open
+    if (trueOrFalse) {
+      setCount(count + 1);
+    } else {
+      setCount(count - 1);
+    }
+  };
+
+  useEffect(() => {
+    // Opens the form if at least one option is checked
+    // Closes the form only if both options are unchecked
+    if (count > 0) {
```

Comment:
couldn't this just a be a boolean if `expandForm` is called for `Expose SQL lab`